core/src/main/scala-3/cats/derived/DerivedContravariant.scala

```diff
@@ -0,0 +1,35 @@
+package cats.derived
+
+import cats.{Contravariant, Functor}
+import shapeless3.deriving.{Const, K1}
+
+import scala.annotation.implicitNotFound
+import scala.compiletime.*
+
+@implicitNotFound("""Could not derive an instance of Contravariant[F] where F = ${F}.
+Make sure that F[_] satisfies one of the following conditions:
+  * it is a constant type [x] =>> T
+  * it is a nested type [x] =>> G[H[x]] where G: Functor and H: Contravariant
+  * it is a generic case class where all fields have a Contravariant instance
+  * it is a generic sealed trait where all subclasses have a Contravariant instance""")
+type DerivedContravariant[F[_]] = Derived[Contravariant[F]]
+object DerivedContravariant:
+  type Or[F[_]] = Derived.Or[Contravariant[F]]
+  inline def apply[F[_]]: Contravariant[F] =
+    import DerivedContravariant.given
+    summonInline[DerivedContravariant[F]].instance
+
+  given [T]: DerivedContravariant[Const[T]] = new Contravariant[Const[T]]:
+    def contramap[A, B](fa: T)(f: B => A): T = fa
+
+  given [F[_], G[_]](using F: DerivedFunctor.Or[F], G: Or[G]): DerivedContravariant[[x] =>> F[G[x]]] =
```

Comment:
I wonder if we can now support the other composition as well which is `F: Contravariant` and `G: Functor` - I think it was ambiguous in Scala 2.

Reply:
Yeah I was just doing `Invariant` and realized there are missing cases. I assume so 👍

Reply:
Literally came here to check if that instance existed and found your comment 😂

Reply:
What do you mean that `Invariant` needs the same treatment?
About making a separate issue - feel free 👍

Reply:
Yeah, there are two more instances we could define for `Invariant`. Opened https://github.com/typelevel/kittens/issues/479 to track it